cbcbeat/splittingsolver.py

```diff
@@ -75,6 +79,25 @@
     pass
 
 
+class Merger:
+    def __init__(self, ode_space, vur_space):
```

Comment:
Could you add some documentation of what the `Merger` class does ( it currently does not use any of its input arguments. Should it check that the input functions are in the appropriate space? 
Should it maybe be called `Interpolater` instead, as it is just interpolating from one space to another?
You could also use the `create_transfer_matrix` to generate the interpolation matrix once, if this action is repeated many times:
https://bitbucket.org/fenics-project/dolfin/src/bd54183ed40f3597fe1187499c79c54eb4759f6d/dolfin/fem/PETScDMCollection.h?at=master#PETScDMCollection.h-69

Reply:
Hmm, you are right that we actually don't use the arguments in the constructer. I think we could just get rid of them. This is basically just a hack to make it possible to use the same API as `FunctionAssigner`. I haven't really tested this properly so we might run into som problems when trying to use this in `simcardems`, but I think we can deal with it then. 

That is a great tip to use the `create_transfer_matrix`. Didn't know about that one.  Is this also available through the python API, or only C++? I am thinking that this could also be very useful when transferring variables between the fine and coarse mesh in `simcardems`.

Reply:
It's in both Python and C++ (https://bitbucket.org/fenics-project/dolfin/src/bd54183ed40f3597fe1187499c79c54eb4759f6d/python/test/unit/fem/test_petsc_transfer_matrix.py?at=master#test_petsc_transfer_matrix.py-37,60,75,93,113) I just referred to the C++ one as it is the one that has documentation.



> Could you add some documentation of what the `Merger` class does ( it currently does not use any of its input arguments. Should it check that the input functions are in the appropriate space? Should it maybe be called `Interpolater` instead, as it is just interpolating from one space to another? You could also use the `create_transfer_matrix` to generate the interpolation matrix once, if this action is repeated many times: https://bitbucket.org/fenics-project/dolfin/src/bd54183ed40f3597fe1187499c79c54eb4759f6d/dolfin/fem/PETScDMCollection.h?at=master#PETScDMCollection.h-69

Yes, it could be very useful there as well. Bear in mind that what we save in work for creating the local interpolation, comes with the increased memory cost of another sparse matrix, so if one runs into memory issues, using normal interpolation might be the better solution.